sys/arduino/base.cpp

```diff
@@ -58,3 +63,40 @@ void delay(unsigned long msec)
 {
     xtimer_usleep(1000 * msec);
 }
+
+/*
+ * Bitfield for the state of the ADC-channels.
+ * 0: Not initialized
+ * 1: Successfully initialized
+ */
+ static uint16_t adc_line_state = 0;
+
+int analogRead(int arduino_pin)
+{
+    int adc_value;
+
+    /* Check if the ADC line is valid */
+    assert((arduino_pin >= 0) && (arduino_pin < ANALOG_PIN_NUMOF));
+
+    /* Initialization of given ADC channel */
+    if (!(adc_line_state & (1 << arduino_pin))) {
+        if (adc_init(ADC_LINE(arduino_analog_map[arduino_pin])) == 0) {
```

Comment:
Also: drop the `ADC_LINE` here! `arduino_analog_map` already consists of entries of type `adc_t`...